encoding/codecv3/codecv3.go

```diff
@@ -0,0 +1,263 @@
+package codecv3
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"fmt"
+
+	"github.com/scroll-tech/go-ethereum/accounts/abi"
+	"github.com/scroll-tech/go-ethereum/common"
+	"github.com/scroll-tech/go-ethereum/crypto"
+	"github.com/scroll-tech/go-ethereum/crypto/kzg4844"
+
+	"github.com/scroll-tech/da-codec/encoding"
+	"github.com/scroll-tech/da-codec/encoding/codecv2"
+)
+
+// MaxNumChunks is the maximum number of chunks that a batch can contain.
+const MaxNumChunks = codecv2.MaxNumChunks
+
+// DABlock represents a Data Availability Block.
+type DABlock = codecv2.DABlock
+
+// DAChunk groups consecutive DABlocks with their transactions.
+type DAChunk = codecv2.DAChunk
+
+// DABatch contains metadata about a batch of DAChunks.
+type DABatch struct {
+	// header
+	Version              uint8
+	BatchIndex           uint64
+	L1MessagePopped      uint64
+	TotalL1MessagePopped uint64
+	DataHash             common.Hash
+	BlobVersionedHash    common.Hash
+	ParentBatchHash      common.Hash
+	LastBlockTimestamp   uint64
+	BlobDataProof        [64]byte
+
```

Comment:
About removing the skipped bitmap: It's technically still part of the batch (we provide it in `commitBatch`) but not part of the batch header and batch hash.

I'm fine with removing it here, but how will you handle this in `rollup-relayer`? Just call `encoding.ConstructSkipBitmap` directly?

Reply:
Yes. I will call `encoding.ConstructSkipBitmap` to get it ad-hoc, since it's low frequency and bitmap is no longer part of batch header encoding now.